### PR TITLE
Fix compile error with 3.8

### PR DESCRIPTION
**Pull request by Dormouse759**
This resolves compile errors with Python 3.8 on the 0.29.x branch.

### CODE REVIEW

**Comment by Dormouse759** [comment]
Resolves #2938

**Comment by scoder** [comment]
Thanks!

**Comment by vstinner** [comment]
Oh cool, well done @Dormouse759.

@scoder @richard-scott @mgedmin: Would it be to get a Cython 0.29.8 release with this fix, please?

FYI we already have the fix downstream in Fedora, but we really love to get fixes merged upstream ;-)

Note: @Dormouse759, @hroncok and me are working for Red Hat in Python Maintenance team (Fedora and RHEL). It's a team work ;-)

**Comment by mgedmin** [comment]
I see a https://github.com/cython/cython/releases/tag/0.29.8 that is 18 hours old, so presumably we're just waiting for the release machinery to do all the builds and upload them?

Or maybe the release machinery is broken: https://travis-ci.org/cython/cython/builds/537945371.  The Python 3 MacOS build broke.  Wow the log is huge.  It ends in

> The job exceeded the maximum time limit for jobs, and has been terminated.

The job run time was 50 minutes.  (The Python 2 MacOS build finished in 18 minutes.)

**Comment by scoder** [comment]
0.29.8 was released with this fix included.

>  The Python 3 MacOS build broke

Happens. Random hickup in travis.

File: Cython/Utility/Profile.c [hunk 1]
```diff
@@ -293,20 +293,18 @@ static PyCodeObject *__Pyx_createFrameCodeObject(const char *funcname, const cha
     PyObject *py_funcname = 0;
     PyCodeObject *py_code = 0;
 
-    #if PY_MAJOR_VERSION < 3
+#if PY_MAJOR_VERSION >= 3
+    py_code = PyCode_NewEmpty(srcfile, funcname, firstlineno);
```
Comment (scoder) [inline Cython/Utility/Profile.c]: Argh!
I totally missed the fact that this entirely lacks error handling. :(
Time for a bug fix release, I guess…

**Comment by scoder** [comment]
Ok, not a random hickup. I released 0.29.9 to fix that.

**Pull request merged**